Mathlib/Data/ZMod/Quotient.lean

```diff
@@ -203,21 +200,32 @@ open Subgroup
 
 variable {α : Type*} [Group α] (a : α)
 
-/-- See also `orderOf_eq_card_zpowers`. -/
-@[to_additive add_order_eq_card_zmultiples' "See also `add_order_eq_card_zmultiples`."]
-theorem order_eq_card_zpowers' : orderOf a = Nat.card (zpowers a) := by
+/-- See also `Fintype.card_zpowers`. -/
+@[to_additive Nat.card_zmultiples "See also `Fintype.card_zmultiples`."]
+theorem Nat.card_zpowers : Nat.card (zpowers a) = orderOf a := by
   have := Nat.card_congr (MulAction.orbitZpowersEquiv a (1 : α))
-  rwa [Nat.card_zmod, orbit_subgroup_one_eq_self, eq_comm] at this
-#align order_eq_card_zpowers' order_eq_card_zpowers'
-#align add_order_eq_card_zmultiples' add_order_eq_card_zmultiples'
+  rwa [Nat.card_zmod, orbit_subgroup_one_eq_self] at this
+#align order_eq_card_zpowers' Nat.card_zpowers
+#align add_order_eq_card_zmultiples' Nat.card_zmultiples
 
 variable {a}
 
+@[to_additive (attr := simp) finite_zmultiples]
+lemma finite_zpowers : (zpowers a : Set α).Finite ↔ IsOfFinOrder a := by
+  simp only [← orderOf_pos_iff, ← Nat.card_zpowers, Nat.card_pos_iff, ← SetLike.coe_sort_coe,
+    nonempty_coe_sort, Nat.card_pos_iff, Set.finite_coe_iff, Subgroup.nonempty, true_and]
+
+@[to_additive (attr := simp) infinite_zmultiples]
+lemma infinite_zpowers : (zpowers a : Set α).Infinite ↔ ¬IsOfFinOrder a := finite_zpowers.not
+
 @[to_additive IsOfFinAddOrder.finite_zmultiples]
-theorem IsOfFinOrder.finite_zpowers (h : IsOfFinOrder a) : Finite <| zpowers a := by
-  rw [← orderOf_pos_iff, order_eq_card_zpowers'] at h
-  exact Nat.finite_of_card_ne_zero h.ne.symm
+protected alias ⟨_, IsOfFinOrder.finite_zpowers⟩ := finite_zpowers
 #align is_of_fin_order.finite_zpowers IsOfFinOrder.finite_zpowers
 #align is_of_fin_add_order.finite_zmultiples IsOfFinAddOrder.finite_zmultiples
 
+/-- The cardinality of a subgroup is greater than the order of any of its elements. -/
+@[to_additive "The cardinality of a subgroup is greater than the order of any of its elements."]
+lemma Subgroup.orderOf_le_card (s : Subgroup α) (hs : (s : Set α).Finite) (ha : a ∈ s) :
+    orderOf a ≤ Nat.card s := by rw [← Nat.card_zpowers]; exact Nat.card_mono hs $ zpowers_le.2 ha
```

Comment:
There's Cauchy's theorem [orderOf_dvd_nat_card](https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/OrderOfElement.html#orderOf_dvd_nat_card) which doesn't require the `Finite` hypothesis, and of course if `Nat.card s` is nonzero then dividing it implies being smaller than or equal to it.

Reply:
Are you saying you don't want my lemma to exist? I don't really mind either way, but I do feel like this is a sensible statement to look for.

Reply:
@alreadydone, can you check whether you're happy wiith my new statement before I hit merge?

Reply:
Sorry, it's (a special case of) Lagrange's theorem ... I think the LE version is fine if you find it more convenient, and just wanted to point out it follows from the Dvd version (does it requires more imports though? Your last commit has a large diff and build error) and that you should add the Group version too.

Reply:
What group version do you have in mind? I assume you meant submonoid version, rather?

Reply:
I mean the version of [orderOf_dvd_nat_card](https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/OrderOfElement.html#orderOf_dvd_nat_card) with dvd replaced by le (assuming `[Finite G]`). The submonoid version also seems to be a good idea, since Lagrange's theorem doesn't hold there so there's no dvd version ...

Reply:
Okay, I've restored my lemma and added the submonoid version.